### PR TITLE
v0.1.0

### PR DESCRIPTION
**Pull request by jlalmes**

### CODE REVIEW

**Comment by juansalvatore** [comment]
hey! @jlalmes is there a reason this pr wasn't merged to master? 
btw great work 💪

File: src/link.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+            );
+            isDone = true;
+          },
+        },
+      };
+
+      port.postMessage({
+        trpc: {
+          id,
+          jsonrpc: undefined,
```
Comment (juansalvatore) [inline src/link.ts]: why is this hardcoded as undefined?

**Comment by jlalmes** [comment]
Hi @juansalvatore, I wanted to make a few more modifications before publishing:

1. Support [tRPC v10](https://trpc.io/docs/v10/).
2. Support [Subscriptions](https://trpc.io/docs/v10/subscriptions).
3. Make bi-directional connections easier & more generic, using [`webext-bridge`](https://github.com/zikaari/webext-bridge) or similar i.e: 
   * webpage 👉 service-worker (using content-script as proxy)
   * webpage 👉 content-script
   * content-script 👉 webpage
   * content-script 👉 service-worker
   * service-worker 👉 content-script
   * service-worker 👉 webpage (using content-script as proxy)

**Comment by nadilas** [comment]
> Hi @juansalvatore, I wanted to make a few more modifications before publishing:
> 
> 1. Support [tRPC v10](https://trpc.io/docs/v10/).
> 2. Support [Subscriptions](https://trpc.io/docs/v10/subscriptions).
> 3. Make bi-directional connections easier & more generic, using [`webext-bridge`](https://github.com/zikaari/webext-bridge) or similar i.e:
>    
>    * webpage 👉 service-worker (using content-script as proxy)
>    * webpage 👉 content-script
>    * content-script 👉 webpage
>    * content-script 👉 service-worker
>    * service-worker 👉 content-script
>    * service-worker 👉 webpage (using content-script as proxy)

If you are thinking of making it more generic, I’d be interested in helping to make a plasmo plugin similar to https://github.com/PlasmoHQ/storage
I believe it would be a nice fit

**Comment by jlalmes** [comment]
> make a plasmo plugin

Hi @nadilas, this sounds like an interesting idea! What do you imagine this API looking like? 

I will admit that I have tried out Plasmo before but found it to be too opinionated for my use-cases at this early stage of its development. For this reason I would like `trpc-chrome` to be decoupled from Plasmo, but a plugin (i.e. `@plasmohq/trpc`) that has this package as a dependency with additional Plasmo specific behaviour could be a great middle-ground.

**Comment by louisgv** [comment]
> [...] I would like trpc-chrome to be decoupled from Plasmo, but a plugin (i.e. @plasmohq/trpc) that has this package as a dependency with additional Plasmo specific behaviour could be a great middle-ground.

I totally agree - trpc-chrome should be a standalone library similar to web-ext-bridge. I don't think there's any advantage/utility for it to be coupled with Plasmo. Plasmo itself is more of a compiler/precompiler/bundler, it doesn't do much at runtime (besides HMR injection etc...) unlike this library which is runtime. 

Plasmo offload the runtime framework to companion dependency like `plasmohq/storage` and `/rpc` so IMO it's the optimal-ground not a middle-ground at all xD

> I will admit that I have tried out Plasmo before but found it to be too opinionated for my use-cases at this early stage of its development. 

Side note: What'd you like to see in Plasmo v1.0? I'd love to hear more about your use case and how the framework can help accommodate it :) 

I'm looking through the trpc PR and based on my thinking, the key value add that a `@plasmohq/trpc` can provide is a reactive hook API for those who want to use this with Plasmo. The more abstraction over the base API for the front-end, the more power to the extension developer.

**Comment by nadilas** [comment]
Hi @jlalmes, I haven't put any thought into how such an interface would look like, it was a spur of the moment idea so far. I'll give it some thought and come back with a proposal. Any wish where I should post it?

I do agree that a dedicated `@plasmohq/trpc` would be more appropriate 👍 

At first glance it seemed to me that `createChromeHandler` could be conforming to a cross-platform interface, which would enable code re-use for the "next" package.

**Pull request merged**